portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -80,6 +104,31 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     datastore.put(commentEntity);
 
-    response.sendRedirect("/comments.html");
+    String url = "/comments.html?numComments=" + numDisplay + "&commentSort=" + sortOrder;
+
+    response.sendRedirect(url);
+  }
+
+  /**
+   * Gets the sort order from the request header, and parses it accordingly.
+   * @return an integer indicating the sort order.
+   */
+  private int getSortOrder(HttpServletRequest request) {
+    String data = request.getHeader("commentSort");
+    if ("timeAscending".equals(data)) {
```

Comment:
This is basically a Yoda condition but without the benefit of avoiding assignment
(https://en.wikipedia.org/wiki/Yoda_conditions#:~:text=In%20programming%20jargon%2C%20Yoda%20conditions,side%20of%20the%20conditional%20statement)

You should reverse the order of the arguments.

Reply:
I did this mainly to avoid null pointer exceptions, especially if commentSort doesn't have a value assigned to it. At least during the early stages of developing this feature, I tended to get a lot of nullpointerexceptions. I will switch and do more thorough tests to see if this is no longer the case.